build-logic/binary-compatibility/src/test/kotlin/gradlebuild/binarycompatibility/AbstractBinaryCompatibilityTest.kt

```diff
@@ -380,33 +381,43 @@ abstract class AbstractBinaryCompatibilityTest {
         }
 
         fun assertHasErrors(vararg errors: String) {
-            assertThat("Has errors", richReport.errors.map { it.message }, CoreMatchers.equalTo(errors.toList()))
+            assertThat("Has errors", richReport.errors.map { it.message }, inAnyOrder(errors))
```

Comment:
ℹ️  We are now matching items in any order.

I believe this does not take away from the assertion itself, as it does not matter much in which order the items appear in the report. We are still checking that each and every item is present, though.

The benefit of the change is in the developer experience, when it comes to resolving errors in the tests using these assertions. The `Matchers.containsInAnyOrder` matcher provides a much clearer description of is missing or unexpected, which is very useful for the number of items we are dealing with in those tests.